### PR TITLE
Renaming nginx/fulltest-provider -> nginx/fulltest & delete old fulltest dir.

### PR DESCRIPTION
**Pull request by bhess**
As suggested in https://github.com/open-quantum-safe/oqs-demos/pull/245#issuecomment-1792653164:

Deletes the old OpenSSL 1.1.1 nginx "fulltest" and renames the oqs-provider based "fulltest-provider" to "fulltest".

### CODE REVIEW

File: nginx/fulltest/README.md [hunk 1]
```diff
@@ -4,45 +4,40 @@ This folder contains all scripts to [build a QSC-enabled nginx server running on
...
+#### Activation
 
-#### First-time execution
+Execute `OPENSSL_CONF=/opt/openssl/.openssl/ssl/openssl.cnf /opt/nginx/sbin/nginx -c /opt/nginx/interop.conf` to start the test server.
 
-`python3 genconfig.py` generates all required QSC certificates. Execute this at the first installation and/or if/when algorithms supported by liboqs have changed since the last installation. **This script overwrites an existing installation's configuration files. Use with care on a live server.** Be sure to have the directory `pki` available/created: For example environmental setup, see e.g., [package.sh](package.sh).
+*Note*: From nginx version 1.25.2, nginx does not try to load OpenSSL configuration if the --with-openssl option was used to built OpenSSL. We therefore have to set the `OPENSSL_CONF` environment variable when activating nginx.
 
-#### Activation
+*Note*: As the server many of ports, the server may need to be configured to permit this, e.g., using `ulimit -S -n 4096`.
```
Comment (baentsch) [inline nginx/fulltest/README.md:39]: typo? "As the server may open many ports..."

File: nginx/fulltest/README.md [hunk 2]
```diff
@@ -4,45 +4,40 @@ This folder contains all scripts to [build a QSC-enabled nginx server running on
...
 
-Execute `/opt/nginx/sbin/nginx -c /opt/nginx/interop.conf` to start the test server.
+#### Test run
 
-*Note*: As the server opens thousands of ports, the server may need to be configured to permit this, e.g., using `ulimit -S -n 4096`.
+The `testrun.sh` script runs test connections against all ports configured by the server. To run the script, execute `testrun.sh openquantumsafe/curl`.
```
Comment (baentsch) [inline nginx/fulltest/README.md:43]: Add reference to proper version (of `liboqs` and docker image)?

File: nginx/fulltest/genconfig.py [hunk 3]
```diff
@@ -114,23 +141,27 @@ def write_nginx_config(f, i, cf, port, sig, k):
 
            # deactivate if you don't like tables:
            i.write("<tr><td>"+sig+"</td><td>"+k+"</td><td>"+str(port)+"</td><td><a href=https://"+TESTFQDN+":"+str(port)+">"+sig+"/"+k+"</a></td></tr>\n")
-           if k in chromium_algs and not ("_" in sig and (sig.startswith("p") or (sig.startswith("rsa")))):
-               cf.write("<tr><td>"+sig+"</td><td>"+k+"</td><td>"+str(port)+"</td><td><a href=https://"+TESTFQDN+":"+str(port)+">"+sig+"/"+k+"</a></td></tr>\n")
+           # chromium support discontinued
```
Comment (baentsch) [inline nginx/fulltest/genconfig.py:144]: Is this correct given https://github.com/open-quantum-safe/oqs-demos/pull/245? @pi-314159 : What's your take? Should we completely remove explicit reference to Chromium?

File: nginx/fulltest/genconfig.py [hunk 4]
```diff
@@ -114,23 +141,27 @@ def write_nginx_config(f, i, cf, port, sig, k):
...
+   cf = None
+   #cf = open(chromiumfilename, "w")
    # copy baseline templates
    with open(TEMPLATE_FILE, "r") as tf:
      for line in tf:
        i.write(line)
-   with open(CHROMIUM_TEMPLATE_FILE, "r") as ctf:
-     for line in ctf:
-       cf.write(line)
+   #chromium support discontinued
```
Comment (baentsch) [inline nginx/fulltest/genconfig.py:161]: See comment above. OK to completely remove Chromium support?

**Review by baentsch — APPROVED** [review]
Conceptually LGTM -- two nits and one question regarding Chromium support, though.

File: nginx/fulltest/genconfig.py (hunk 3 above)
Comment (bhess) [reply nginx/fulltest/genconfig.py:144]: We wanted to not include an explicit support list anymore for Chromium, but I think the statement "# chromium support discontinued" is incorrect with the work by @pi-314159.
I'd be in favor to update the genconfig.py file by removing references to Chromium (particularly that chromium support is discontinued).

File: nginx/fulltest/README.md (hunk 1 above)
Comment (bhess) [reply nginx/fulltest/README.md:39]: Thanks for the catch, will update.

File: nginx/fulltest/README.md (hunk 2 above)
Comment (bhess) [reply nginx/fulltest/README.md:43]: Will update with a reference to the tag to be used.

**Pull request merged**